### PR TITLE
[ML] Add multimodal distribution field processing for anomaly score explanation

### PR DESCRIPTION
**Pull request by valeriy42**
The companion PR to elastic/ml-cpp#2440 adds processing of `multimodal_distribution` field in the anomaly score explanation. I added a changelog entry in the `ml-cpp` PR hence I mark this PR as a non-issue.

### CODE REVIEW

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyScoreExplanation.java [hunk 1]
```diff
@@ -78,6 +81,7 @@ public AnomalyScoreExplanation(StreamInput in) throws IOException {
         this.upperConfidenceBound = in.readOptionalDouble();
         this.highVariancePenalty = in.readOptionalBoolean();
         this.incompleteBucketPenalty = in.readOptionalBoolean();
+        this.multimodalDistribution = in.readOptionalBoolean();
```
Comment (przemekwitek) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyScoreExplanation.java:84]: Should this be guarded with a version check? I.e.:

```
if (in.getVersion().onOrAfter(Version.V_8_7_0) {
   ...
}
```

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyScoreExplanation.java [hunk 2]
```diff
@@ -92,6 +96,7 @@ public void writeTo(StreamOutput out) throws IOException {
         out.writeOptionalDouble(upperConfidenceBound);
         out.writeOptionalBoolean(highVariancePenalty);
         out.writeOptionalBoolean(incompleteBucketPenalty);
+        out.writeOptionalBoolean(multimodalDistribution);
```
Comment (przemekwitek) [inline x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyScoreExplanation.java:99]: Same concern here...

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyScoreExplanation.java (hunk 1 above)
Comment (valeriy42) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyScoreExplanation.java]: Good spot. Fixed it in 562e992

File: x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyScoreExplanation.java (hunk 2 above)
Comment (valeriy42) [reply x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/results/AnomalyScoreExplanation.java]: Fixed in 562e992

**Comment by valeriy42** [comment]
Thank you for your comments, @przemekwitek. Can you please take another look?

**Review by przemekwitek — APPROVED** [review]
LGTM

**Pull request merged**